### PR TITLE
Replaced ListView with RecyclerView

### PR DESCRIPTION
**Pull request by Smarshal21**
### 🖼️ Screenshots
**Portrait**

**Landscape**

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [x] 🔖 Capability is checked or not needed 
- [x] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [x] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

**Comment by Smarshal21** [comment]
@AndyScherzinger  PTAL and do tell me if LiveData should be implemented.

**Review by AndyScherzinger — APPROVED** [review]

**Comment by AndyScherzinger** [comment]
> @AndyScherzinger PTAL and do tell me if LiveData should be implemented.

Looks good to me. `LiveData` would be nice if you have the time and motivation for it. I'll take care of merging this PR, so LiveData can/should then go into a separate PR, to keep PRs at a small size, to make it easier for reviewers 👍 

Nice work! 🚀

**Comment by Smarshal21** [comment]
> > @AndyScherzinger PTAL and do tell me if LiveData should be implemented.
> 
> Looks good to me. `LiveData` would be nice if you have the time and motivation for it. I'll take care of merging this PR, so LiveData can/should then go into a separate PR, to keep PRs at a small size, to make it easier for reviewers 👍
> 
> Nice work! 🚀

Sure will do

**Comment by rapterjet2004** [comment]
Looks good! The adapter works great but the `OnClickListener` for the `GeocodingAdapter` items don't seem to be working on this branch. Upon clicking the item, it should take you back to the `LocationPickerActivity` with the new location.

## Master

[replaced-listView-branch-Master.webm](https://github.com/nextcloud/talk-android/assets/69230048/a0cad65d-5e86-40c5-ad3c-7a10cb12205e)

## RecyclerView
[replace-listview-branch-RecyclerView.webm](https://github.com/nextcloud/talk-android/assets/69230048/50fa6391-2d60-4c04-9836-7f04e1666b00)

File: app/src/main/java/com/nextcloud/talk/location/GeocodingActivity.kt [hunk 1]
```diff
@@ -108,16 +114,17 @@ class GeocodingActivity :
...
+            override fun onItemClick(position: Int) {
+                val address: Address = adapter.getItem(position) as Address
+                val geocodingResult = GeocodingResult(address.latitude, address.longitude, address.displayName)
+                val intent = Intent(this@GeocodingActivity, LocationPickerActivity::class.java)
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                intent.putExtra(BundleKeys.KEY_ROOM_TOKEN, roomToken)
+                intent.putExtra(BundleKeys.KEY_GEOCODING_RESULT, geocodingResult)
+                startActivity(intent)
+            }
+        })
```
Comment (rapterjet2004) [inline app/src/main/java/com/nextcloud/talk/location/GeocodingActivity.kt:127]:
``` 
adapter.setOnItemClickListener(object : GeocodingAdapter.OnItemClickListener {
            override fun onItemClick(position: Int) {
                val address: Address = adapter.getItem(position) as Address
                val geocodingResult = GeocodingResult(address.latitude, address.longitude, address.displayName)
                val intent = Intent(this@GeocodingActivity, LocationPickerActivity::class.java)
                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                intent.putExtra(BundleKeys.KEY_ROOM_TOKEN, roomToken)
                intent.putExtra(BundleKeys.KEY_GEOCODING_RESULT, geocodingResult)
                startActivity(intent)
            }
      })
```

move this into the `initAdapter` function, before the `binding.geocodingResults.adapter = adapter`. Currently, the adapter is not being updated after the ItemClicker is set up.

**Review by rapterjet2004 — CHANGES_REQUESTED** [review]
I figured out the bug, the adapter is not being updated after the ItemClicker is set up. Just implement the changes I suggested and it should work. Overall good work!

File: app/src/main/java/com/nextcloud/talk/location/GeocodingActivity.kt (hunk 1 above)
Comment (Smarshal21) [reply app/src/main/java/com/nextcloud/talk/location/GeocodingActivity.kt:127]: Nice catch. 
Thank you @rapterjet2004 for the fix 😁

Comment (Smarshal21) [reply app/src/main/java/com/nextcloud/talk/location/GeocodingActivity.kt:127]: Resolved the issue

**Comment by AndyScherzinger** [comment]
Nice catch @rapterjet2004 👍

**Review by rapterjet2004 — APPROVED** [review]

**Pull request merged**